File: meeting/solutions/meeting.cpp
#include <bits/stdc++.h>
using namespace std;

bool compare(const vector<int> &a, const vector<int> &b) { return a[0] < b[0]; }

vector<vector<int>> reschedule(vector<vector<int>> &schedules,
                               vector<int> &urgent_meeting) {
  vector<vector<int>> result;
  schedules.push_back(urgent_meeting);
  sort(schedules.begin(), schedules.end(), compare);

  for (vector<int> schedule : schedules) {
    if (result.empty() || result.back()[1] < schedule[0]) {
      result.push_back(schedule);
    } else {
      result.back()[1] = max(result.back()[1], schedule[1]);
    }
  }

  return result;
}

int main() {
  int n, start, end;
  vector<vector<int>> schedules;
  vector<int> urgent_meeting(2);

  cin >> n;
  while (n--) {
    cin >> start >> end;
    schedules.push_back({start, end});
  }
  cin >> urgent_meeting[0] >> urgent_meeting[1];

  vector<vector<int>> result = reschedule(schedules, urgent_meeting);
  for (vector<int> schedule : result) {
    cout << schedule[0] << " " << schedule[1] << endl;
  }
}

File: meeting/solutions/meeting.js
process.stdin.resume();
process.stdin.setEncoding("utf-8");

let inputString = "";
let currentLine = 0;

process.stdin.on("data", (inputStdin) => {
    inputString += inputStdin;
});

process.stdin.on("end", () => {
    inputString = inputString.split("\n");
    main();
});

function readLine() {
    return inputString[currentLine++];
}

/**
 * @param {number[][]} schedules - An array of existing meeting schedules, where each schedule is represented as [start, end] pair.
 * @param {number[]} urgent_meeting - The urgent meeting to be inserted, represented as [start, end] pair.
 * @returns {number[][]} The updated schedule with merged intervals, where each element is an [start, end] pair representing a meeting.
 */
function reschedule(schedules, urgent_meeting) {
    schedules.push(urgent_meeting);
    schedules = schedules.sort((a, b) => a[0] - b[0]);

    const result = [];
    for (const schedule of schedules) {
        if (!result.length || result[result.length - 1][1] < schedule[0]) {
            result.push(schedule);
        } else {
            result[result.length - 1][1] = Math.max(
                result[result.length - 1][1],
                schedule[1],
            );
        }
    }

    return result;
}

function main() {
    const schedules = [];
    const n = parseInt(readLine());
    for (let i = 0; i < n; i++) {
        const [start, end] = readLine().split(" ").map(Number);
        schedules.push([start, end]);
    }

    const [urgentStart, urgentEnd] = readLine().split(" ").map(Number);
    const urgentMeeting = [urgentStart, urgentEnd];

    const result = reschedule(schedules, urgentMeeting);
    for (const schedule of result) {
        console.log(`${schedule[0]} ${schedule[1]}`);
    }
}
